src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -202,6 +204,24 @@ internal int StartSession(string ownerUri, string template, ConnectionInfo connI
             return xeSession.Id;
         }
 
+        private static BaseXEStore CreateXEventStore(ConnectionInfo connInfo, SqlStoreConnection connection)
+        {   
+            BaseXEStore store = null;
+            if (connInfo.IsCloud)
+            {
+                if (DatabaseUtils.IsSystemDatabaseConnection(connInfo.ConnectionDetails.DatabaseName))
+                {
+                    throw new NotSupportedException("Cannot profile Azure system databases");
```

Comment:
should the error message be localized?  I assume this will get displayed to the user.